text_generation/causal_lm/cpp/continuous_batching/library/src/continuous_batching_pipeline.cpp

```diff
@@ -235,16 +235,19 @@ class ContinuousBatchingPipeline::Impl {
             step();
         }
 
-        for (auto& generation : generations) {
+        for (size_t generation_idx = 0; generation_idx < generations.size(); ++generation_idx) {
+            const auto& generation = generations[generation_idx];
+            OPENVINO_ASSERT(generation->get_status() == GenerationStatus::FINISHED);
```

Comment:
In accuracy sample we check for status and report to the user if something went wrong. 
With this line, we can potentially lose finished generations that would be processed in further iterations.

Reply:
removed